test/cmdlineTests/standard_import_evmasm/output.json

```diff
@@ -0,0 +1,575 @@
+  0x04
+    /* \"<stdin>\":146:174   */
+  sstore
+    /* \"<stdin>\":189:204   */
+  dataSize(sub_0.sub_0)
+    /* \"<stdin>\":186:187   */
+  0x05
+    /* \"<stdin>\":179:205   */
+  sstore
+    /* \"<stdin>\":220:237   */
+  dataOffset(sub_0.sub_0.sub_0)
+    /* \"<stdin>\":217:218   */
+  0x06
+    /* \"<stdin>\":210:238   */
+  sstore
+    /* \"<stdin>\":253:268   */
+  dataSize(sub_0.sub_0.sub_0)
+    /* \"<stdin>\":250:251   */
+  0x07
+    /* \"<stdin>\":243:269   */
+  sstore
+    /* \"<stdin>\":284:303   */
+  dataOffset(sub_0.sub_1)
+    /* \"<stdin>\":281:282   */
+  0x08
+    /* \"<stdin>\":274:304   */
+  sstore
+    /* \"<stdin>\":319:336   */
+  dataSize(sub_0.sub_1)
+    /* \"<stdin>\":316:317   */
+  0x09
+    /* \"<stdin>\":309:337   */
+  sstore
+stop
+data_acaf3289d7b601cbd114fb36c4d29c85bbfd5e133f14cb355c3fd8d99367964f 48656c6c6f2c20576f726c6421
+
+sub_0: assembly {
+        /* \"<stdin>\":418:433   */
+      dataOffset(sub_0)
+        /* \"<stdin>\":414:416   */
+      0x0a
+        /* \"<stdin>\":407:434   */
+      sstore
+        /* \"<stdin>\":452:465   */
+      dataSize(sub_0)
+        /* \"<stdin>\":448:450   */
+      0x0b
+        /* \"<stdin>\":441:466   */
+      sstore
+        /* \"<stdin>\":484:499   */
+      dataOffset(sub_1)
+        /* \"<stdin>\":480:482   */
+      0x0c
+        /* \"<stdin>\":473:500   */
+      sstore
+        /* \"<stdin>\":518:531   */
+      dataSize(sub_1)
+        /* \"<stdin>\":514:516   */
+      0x0d
+        /* \"<stdin>\":507:532   */
+      sstore
+        /* \"<stdin>\":550:567   */
+      dataOffset(sub_0.sub_0)
+        /* \"<stdin>\":546:548   */
+      0x0e
+        /* \"<stdin>\":539:568   */
+      sstore
+        /* \"<stdin>\":586:601   */
+      dataSize(sub_0.sub_0)
+        /* \"<stdin>\":582:584   */
+      0x0f
+        /* \"<stdin>\":575:602   */
+      sstore
+    stop
+
+    sub_0: assembly {
+            /* \"<stdin>\":658:673   */
+          dataOffset(sub_0)
+            /* \"<stdin>\":654:656   */
+          0x10
+            /* \"<stdin>\":647:674   */
+          sstore
+            /* \"<stdin>\":694:707   */
+          dataSize(sub_0)
+            /* \"<stdin>\":690:692   */
+          0x11
+            /* \"<stdin>\":683:708   */
+          sstore
+        stop
+
+        sub_0: assembly {
+                /* \"<stdin>\":761:770   */
+              invalid
+        }
+    }
+
+    sub_1: assembly {
+            /* \"<stdin>\":833:842   */
+          invalid
+    }
+}
+",
+                    "bytecode":
+                    {
+                        "functionDebugData": {},
+                        "linkReferences": {},
+                        "object": "<BYTECODE REMOVED>",
+                        "opcodes":"<OPCODES REMOVED>",
+                        "sourceMap":"<SOURCEMAP REMOVED>"
+                    },
+                    "deployedBytecode":
+                    {
+                        "functionDebugData": {},
```

Comment:
`functionDebugData`, `immutableReferences`, `linkReferences` and also `generatedSources` not supported right now, but at least right now I'm not sure with `functionDebugData` (need to check semantic) & `immutableReferences` (not tired yet, maybe it just works).. however, I saw that `generatedSources` is just null here, so I use `util::removeNullMembers` on the result.

Reply:
Well, we don't have access to the source anymore when assembling so you cannot get `generatedSources` from the compiler. At the assembling stage the generated sources are considered just another source file (and listed in `sourceList` if present).

`functionDebugData` also does not look like something recoverable from the assembly itself.

`linkReferences` should be supported. You need those to actually link the code. The offsets are produced during assembling and the hash can be calculated from the library name included in the `PUSHLIB` item. So you can probably easily get this from the stack, you may just have to add a function to access it.

Not sure about `immutableReferences`, but it seems like we might not have all the data for that in the assembly. I.e. `immutableReferences` contains names of the variables and their offsets. Offsets you can probably calculate, but I see that names are neither in `assignImmutable()` in the text form nor in `ASSIGNIMMUTABLE` entries in JSON.

Reply:
I think that `functionDebugData` or `immutableReferences` should not appear in the output at all if we can't produce them.

Reply:
I checked how it looks in Yul and apparently we output empty `functionDebugData` there too. In that case I guess no point in doing anything special here. Let's keep it as is.

And `immutableReferences` does work like you mentioned so that's fine too.